site/docs/cow-and-mor.md

```diff
@@ -0,0 +1,195 @@
+ - limitations under the License.
+ -->
+
+# Copy-on-Write and Merge-on-Read
+
+This page explains the concept of copy-on-write and merge-on-read in the context of Iceberg to provide readers more clarity around Iceberg's table spec design.
+
+## Introduction
+
+In Iceberg, copy-on-write and merge-on-read are different ways to handle row-level update and delete operations. Here are their definitions:
+
+- **copy-on-write (CoW)**: an update/delete directly rewrites the entire affected data files.
+- **merge-on-read (MoR)**: update/delete information is encoded in the form of delete files. The table reader can apply all delete information at read time. A compaction process takes care of merging delete files into data files asynchronously. 
+
+Clearly, CoW is more efficient in reading data, but MoR is more efficient in writing data.
+Users can choose to use **BOTH** CoW and MoR against the same Iceberg table based on different situations. 
+A common example is that, for a time-partitioned table, newer partitions with more frequent updates are maintained in the MoR approach through a CDC streaming pipeline,
+and older partitions are maintained in the CoW way with less frequent GDPR updates from batch ETL jobs.
+
+## Copy-on-write
+
+As the definition states, given a user's update/delete requirement, the CoW write process would search for all the affected data files and perform rewrite.
+Spark supports CoW `DELETE`, `UPDATE` and `MERGE` operations through Spark extensions. More details can be found in [Spark Writes](../spark-writes) page.
+
+## Merge-on-read
+
+In the next few sections, we provide more details around the Iceberg MoR design.
+
+### Row-Level Delete File Spec
+
+As documented in the [Spec](../spec/#row-level-deletes) page, Iceberg supports 2 different types of row-level delete files: **position deletes** and **equality deletes**.
+If you are unfamiliar with these concepts, please read the related sections in the spec for more information before proceeding.
+
+Also note that because row-level delete files are valid Iceberg data files, each file must define the partition it belongs to.
+If the file belongs to `Unpartitioned` (the partition spec has no partition field), then the delete file is called a **global delete**. 
+Otherwise, it is called a **partition delete**.
+
+### MoR Update as Delete + Insert
+
+In Iceberg, update is modeled as a delete with an insert within the same transaction, so there is no concept of an "update file".
+During a MoR write transaction, new data files and delete files are committed with the same sequence number.
+During a MoR read process, delete files are applied to data files of strictly lower sequence numbers.
+This ensures the latest updated rows are displayed to users during a MoR read.
+
+### Delete File Writer
+
+From the end user's perspective, it is very rare that they could directly request deletion of a specific row of a specific file given the abstraction provided by Iceberg. 
```

Comment:
I would skip this sentence, not sure it helps too much and I think the examples below are pretty clear.